build-system/tasks/coverage-map/index.js

```diff
@@ -78,26 +81,46 @@ async function autoScroll(page) {
   });
 }
 
+async function htmlTransform() {
+  log(`Transforming ${inputHtml}...`);
+  const transformed = await transform(`examples/${inputHtml}`);
+  const transformedName = `transformed.${inputHtml}`;
+  await fsExtra.ensureDir('dist/transformed');
+  await fs.writeFile(`dist/transformed/${transformedName}`, transformed);
+  log(
+    `Transformation complete. It can be found at "dist/transformed/${transformedName}".`
+  );
+  testUrl = `http://localhost:${serverPort}/dist/transformed/${transformedName}`;
+}
+
 async function generateMap() {
   log(
     `Generating heat map in dist/${outHtml} of ${inputJs}, based on ${coverageJsonName}...`
   );
   await explore(`dist/${inputJs}`, {
-    output: {format: 'html', filename: `${outHtml}`},
+    output: {format: 'html', filename: `dist/${outHtml}`},
     coverage: `dist/${coverageJsonName}`,
     onlyMapped: true,
   });
 }
 
 async function coverageMap() {
   installPackages(__dirname);
+  buildNewServer();
+
+  if (!argv.nobuild) {
+    await dist();
+  }
 
   puppeteer = require('puppeteer');
   explore = require('source-map-explorer').explore;
+  transform = await require('../../server/new-server/transforms/dist/transform')
```

Comment:
This particular `require` needs to wait until `dist()` and `buildNewServer()` finishes.
On initialization, the directory listed does not exist. Furthermore, this `require` will try to access files inside `dist` in the root directory.

Reply:
I see. In that case, would it work to explicitly wait for those actions to complete before invoking the `require` (without the `await`)?

Reply:
Did you mean like wrapping this inside a promise?

Reply:
The function `transform` itself is `async`, and so when called we'll need to `await transform()`, but the transform module is not async and so the `require` call shouldn't be `await`ed. I think what Raghu is suggesting here is that, since this `require` comes after `await dist()` already, by the time this point is reached in the code it should be assumed that `dist/**` exist

Reply:
Given that these packages are all used elsewhere in the module, I'm inclined to say these `require` calls should happen in the functions that use the modules, rather than accessing them via file-level globals. Others may differ in opinions here though.

Reply:
Good point! I've moved them to the corresponding functions.